app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/acquisitiondetails/RebasingEligibilityUtil.scala

```diff
@@ -0,0 +1,138 @@
+      case _               => true
+    }
+
+  def isEligibleForRebase(wasAUkResident: Boolean, assetType: AssetType, purchaseDate: LocalDate): Boolean =
+    (wasAUkResident, assetType, purchaseDate) match {
+      case (true, AssetType.Residential, date) => date.isBefore(RebasingCutoffDates.ukResidents)
+      case (false, AssetType.Residential, date) =>
+        date.isBefore(RebasingCutoffDates.nonUkResidentsResidentialProperty)
+      case (false, AssetType.NonResidential, date) =>
+        date.isBefore(RebasingCutoffDates.nonUkResidentsNonResidentialProperty)
+      case _ => false
+    }
+
+  def isUkAndEligibleForRebase(completeReturn: CompleteReturn): Boolean =
+    completeReturn.triageAnswers.fold(
+      i => i.assetType       -> i.wasAUKResident,
+      c => Some(c.assetType) -> Some(c.countryOfResidence.isUk())
+    ) match {
+      case (Some(assetType), Some(isUkResident)) =>
+        isUkAndEligibleForRebase(isUkResident, completeReturn.acquisitionDetails.acquisitionDate, assetType)
+    }
+
+  def isUkAndEligibleForRebase(
+    wasUkResident: Boolean,
+    acquisitionDate: AcquisitionDate,
+    assetType: AssetType
+  ): Boolean =
+    wasUkResident && rebasingCutOffDate(acquisitionDate, assetType, wasUkResident).isDefined
+
+  def rebasingCutOffDate(
+    acquisitionDate: AcquisitionDate,
+    assetType: AssetType,
+    wasUkResident: Boolean
+  ): Option[LocalDate] = {
+    val cutoffDate =
+      if (wasUkResident)
+        RebasingCutoffDates.ukResidents
+      else if (assetType === AssetType.Residential)
+        RebasingCutoffDates.nonUkResidentsResidentialProperty
+      else RebasingCutoffDates.nonUkResidentsNonResidentialProperty
+
+    Some(cutoffDate).filter(acquisitionDate.value.isBefore)
+  }
+
+  def invalidForRebasing(
+    acquisitionDate: AcquisitionDate,
+    assetType: AssetType,
+    wasUkResident: Boolean
+  ): Option[LocalDate] = {
+    val cutoffDate =
+      if (wasUkResident)
+        RebasingCutoffDates.ukResidents
+      else if (assetType === AssetType.Residential)
+        RebasingCutoffDates.nonUkResidentsResidentialProperty
+      else RebasingCutoffDates.nonUkResidentsNonResidentialProperty
+
+    Some(cutoffDate).filter(acquisitionDate.value.isBefore)
+  }
+
+  def getRebasingCutOffDate(
+    assetType: AssetType,
+    wasUkResident: Boolean
+  ): LocalDate =
+    if (wasUkResident)
+      RebasingCutoffDates.ukResidents
+    else if (assetType === AssetType.Residential)
+      RebasingCutoffDates.nonUkResidentsResidentialProperty
+    else RebasingCutoffDates.nonUkResidentsNonResidentialProperty
+
+  def getRebasingCutOffDate(completeReturn: CompleteReturn): LocalDate =
+    completeReturn.triageAnswers.fold(
+      i => i.assetType       -> i.wasAUKResident,
+      c => Some(c.assetType) -> Some(c.countryOfResidence.isUk())
+    ) match {
+      case (Some(assetType), Some(isUkResident)) =>
+        getRebasingCutOffDate(assetType, isUkResident)
+    }
+
+  def shouldRedirect(
```

Comment:
it's hard to see how this method would be used in any other context than the one place you're using it at the moment - I'd consider narrowing the scope of this a bit to maybe inside the controller with a more appropriate name